Module 3. Components

tor4 opened the pull request.

tor4 committed:
Install Storybook

tor4 committed:
Add new components and stories for them

Stories aer added for existig components.

vadimgordenya commented on src/Components/MovieTile/MovieTile.jsx:
```diff
@@ -0,0 +1,58 @@
+            <div className={`context-menu ${menuOpened && 'opened'}`} onClick={openMenu}>
+                <div className='menu'>
+                    <span className='close' onClick={closeMenu}>x</span>
+                    <span className='option' onClick={(e) => {
```
For buttons we need to use button tag, to avoid accessibility issues in future.

tor4 replied:
Good point

vadimgordenya commented on src/Components/MovieTile/MovieTile.jsx:
```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import { NO_IMAGE_URL } from '../../Utils/constants';
+import './MovieTile.css';
+import { useState } from 'react';
```
Imports ordering: libraries should be on the top level.

vadimgordenya commented on src/Components/SortControl/SortControl.css:
```diff
@@ -0,0 +1,40 @@
+
+.sort-control .custom-select select {
+  text-transform: uppercase;
+  color: #fff;
```
It's better to create separate file with variables for colors/fonts/etc

tor4 replied:
Added several variables to index.css for font-size and color.

vadimgordenya commented on src/Components/SearchForm/SearchForm.stories.js:
```diff
@@ -0,0 +1,28 @@
+    tags: ['autodocs'],
+};
+
+export const Default = {
```
Do we really need it here?

tor4 replied:
Not really

vadimgordenya commented on src/Components/SearchForm/SearchForm.stories.js:
```diff
@@ -0,0 +1,28 @@
+import {SearchForm} from './SearchForm';
+import { userEvent, waitFor, within } from '@storybook/testing-library';
+import { expect } from '@storybook/jest';
+import '../../index.css';
```
The name for the css file should be the same as a component name for more readability.

tor4 replied:
Understand. But this is common css file. Haven't found how to include it for all stories by default.

vadimgordenya commented on src/Components/MovieTile/MovieTile.test.js:
```diff
@@ -0,0 +1,57 @@
+import { render, screen, fireEvent } from "@testing-library/react";
+import { MovieTile } from './MovieTile';
+import { NO_IMAGE_URL } from "../../Utils/constants";
```
Take a look in aliases side to avoid following path: "../../"

tor4 replied:
Could you provide more details here?

tor4 committed:
Fixes after review

- added css variables for font-size and color
- reorder imports
- use button tag for menu items

tor4 commented:
@vadimgordenya I've pushed some changes after review.
